app/javascript/components/tree-view/hierarchical-tree-view.jsx

```diff
@@ -1,72 +1,168 @@
+/* eslint camelcase: ["warn", {allow: ["bs_tree", "tree_name", "click_url", "check_url", "allow_reselect", "hierarchical_check", "silent_activate", "select_node"]}] */
 import React, { useEffect } from 'react';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 import Tree, {
   Node,
-  defaultStore,
-  callBack,
-  ActionTypes,
 } from '@manageiq/react-ui-components/dist/wooden-tree';
+import { http } from '../../http_api';
+import { combineReducers } from '../../helpers/redux';
+import reducers, { ACTIONS } from './reducers/index';
+import basicStore from './reducers/basicStore';
 
-const convertData = data => Tree.convertHierarchicalTree(Tree.initHierarchicalTree(JSON.parse(data)));
+const flatten = data => Tree.convertHierarchicalTree(Tree.initHierarchicalTree(data));
+
+const callBack = (nodeId, type, value, namespace) => ({
+  nodeId, type, value, namespace,
+});
+
+/**
+ * Helper
+ * Changes class to className.
+ * Sets undefined checkbox to null.
+ * Moves the key to attr.key.
+ */
+const convert = tree => tree.map((n) => {
+  let node = { ...n };
+  if (node.class) {
+    node = { ...node, classes: node.class };
+    delete node.class;
+  }
+
+  if (node.state.checked === 'undefined') {
+    node = { ...node, state: { ...node.state, checked: null } };
+  } else if (node.state.checked === null) {
+    node = { ...node, state: { ...node.state, checked: false } };
+  }
+
+  if (node.key) {
+    node = { ...node, attr: { key: node.key } };
+    delete node.key;
+  }
+
+  if (node.nodes) {
+    node = { ...node, nodes: convert(node.nodes) };
+  }
+
+  return node;
+});
+
+/**
+ * Helper
+ * Activates the first active node in the tree, which is passed trought the props.
+ *
+ * FIXME Delete this function when bakcend is sending normal data.
```

Comment:
this comment is not that useful if we don't know how the "normal data" should look
there's a similar comment where this function is used,
but I see no comment about what specifically is wrong or what needs change where.
Can you be more specific please?

Reply:
By normal data he probably means the same format as his component is expecting. As we still have the old trees on the left side rendered via angular, I'm sending everything in the old format. But @himdel's right, this could be described in a more understandable way.